File: tests/playwright/tests/objects/pages/signup.page.ts
import { expect, Locator, Page, test } from '@playwright/test';
import { SignUpForm } from '../forms/auth/signup.form';

export class SignUpPage {
  private readonly page: Page;
  private readonly path: string;
  private readonly container: Locator;
  private readonly _signUpForm: () => SignUpForm;

  constructor(page: Page) {
    this.page = page;
    this.path = '/signup';
    this.container = page.getByTestId('container').filter({ has: page.getByTestId('page-signup') });
    this._signUpForm = () => new SignUpForm(this.container.getByTestId('form-signup'));
  }

  get signUpForm() {
    return this._signUpForm();
  }

  async open() {
    await test.step(`Open 'Sign up' page`, async () => {
      await this.page.goto(this.path);
    });
  }

  async shouldBeOpened() {
    await test.step(`'Sign up' page should be opened`, async () => {
      await expect(this.container).toBeVisible();
      await expect(this.page).snapshot();
    });
  }
}

File: tests/selenide/src/test/java/objects/pages/SignUpPage.java
package objects.pages;

import com.codeborne.selenide.SelenideElement;
import io.qameta.allure.Step;
import objects.forms.SignUpForm;

import static com.codeborne.selenide.Condition.visible;
import static com.codeborne.selenide.Selenide.$x;
import static utils.selectors.ElementSelectors.byTestId;

public class SignUpPage extends BasePage {

    private final SelenideElement container = $x("//*[@data-testid='container' and .//*[@data-testid='form-signup']]");
    private final SignUpForm signUpForm = new SignUpForm(container.$(byTestId("form-signup")));


    public SignUpPage() {
        super("/signup");
    }

    public SignUpForm signUpForm() {
        return signUpForm;
    }

    @Step("'Sign up' page should be opened")
    @Override
    public SignUpPage shouldBeOpen() {
        this.container.shouldBe(visible);
        return this;
    }

    @Step("Open 'Sign up' page")
    public SignUpPage open() {
        super.open();
        return this;
    }

}
